File: PSP-DAM-ACTEVA03B/python_src/cocina.py
import os
from pedido import Pedido
from cocinero import Cocinero
import threading

if __name__ == '__main__':
    base_path = os.getcwd()
    log_path = os.path.join(base_path, 'log_pedidos.txt')

    # inicializar log vacío
    with open(log_path, 'w', encoding='utf-8') as f:
        f.write('')

    lock = threading.Lock()
    pedidos = [
        Pedido(1, 'Lubina a la sal'),
        Pedido(2, 'Filete de ternera'),
        Pedido(3, 'Carne de cerdo a la barbacoa'),
        Pedido(4, 'Vieira gratinada'),
        Pedido(5, 'Langosta a la plancha'),
        Pedido(6, 'Gamba a la gabardina'),
    ]

    cocineros = [
        Cocinero('Cocinero-1', pedidos, lock, log_path),
        Cocinero('Cocinero-2', pedidos, lock, log_path),
        Cocinero('Cocinero-3', pedidos, lock, log_path),
    ]

    for c in cocineros:
        c.start()

    for c in cocineros:
        c.join()

    print('Todos los pedidos han sido procesados.')

File: java_src/Cocina.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardOpenOption;
import java.util.ArrayList;
import java.util.List;

public class Cocina {
    public static void main(String[] args) {
        String basePath = System.getProperty("user.dir");
        String logPath = basePath + File.separator + "log_pedidos.txt";

        // inicializar log vacío en UTF-8
        try {
            Path p = Path.of(logPath);
            Files.write(p, new byte[0], StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
        } catch (IOException e) {
            System.err.println("No se puede crear log: " + e.getMessage());
            return;
        }

        Object lock = new Object();
        List<Pedido> pedidos = new ArrayList<>();
        pedidos.add(new Pedido(1, "Calamares"));
        pedidos.add(new Pedido(2, "Huevos fritos con chorizo"));
        pedidos.add(new Pedido(3, "Lentejas"));
        pedidos.add(new Pedido(4, "Guisantes con jamon"));
        pedidos.add(new Pedido(5, "Cachopo"));
        pedidos.add(new Pedido(6, "Mejillones al vapor"));

        // al menos 3 cocineros
        Cocinero c1 = new Cocinero("Cocinero-1", pedidos, lock, logPath);
        Cocinero c2 = new Cocinero("Cocinero-2", pedidos, lock, logPath);
        Cocinero c3 = new Cocinero("Cocinero-3", pedidos, lock, logPath);

        c1.start();
        c2.start();
        c3.start();

        try {
            c1.join();
            c2.join();
            c3.join();
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }

        System.out.println("Todos los pedidos han sido procesados.");
    }
}